types/seccomp.go

```diff
@@ -2,12 +2,20 @@ package types
 
 // Seccomp represents the config for a seccomp profile for syscall restriction.
 type Seccomp struct {
-	DefaultAction Action     `json:"defaultAction"`
-	Architectures []Arch     `json:"architectures"`
-	Syscalls      []*Syscall `json:"syscalls"`
+	DefaultAction Action         `json:"defaultAction"`
+	Architectures []Arch         `json:"architectures,omitempty"`
+	ArchMap       []Architecture `json:"archMap,omitempty"`
```

Comment:
Yes, it is designed to be fully compatible.

On 31 Aug 2016 13:39, "Vincent Demeester" <notifications@github.com> wrote:

> In types/seccomp.go
> <https://github.com/docker/engine-api/pull/313#discussion_r76979496>:
>
> > @@ -2,12 +2,20 @@ package types
> >
> >  // Seccomp represents the config for a seccomp profile for syscall restriction.
> >  type Seccomp struct {
> > -	DefaultAction Action     `json:"defaultAction"`
> > -	Architectures []Arch     `json:"architectures"`
> > -	Syscalls      []*Syscall `json:"syscalls"`
> > +	DefaultAction Action         `json:"defaultAction"`
> > +	Architectures []Arch         `json:"architectures,omitempty"`
> > +	ArchMap       []Architecture `json:"archMap,omitempty"`
>
> This looks like a bit of duplication (or I don't really get it 😅), but I
> guess this is for retro-compatibility right ? (if yes, a comment about that
> would be perfect 😝 )
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/engine-api/pull/313/files/b4b3edb80a5aabf9567a769e2aca4b6913aed800#r76979496>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAdcPOZaWoWoNEnLRhBuT3MS1VF3alvrks5qlXYegaJpZM4JLbHs>
> .
>